server/room-battle.js

```diff
@@ -434,6 +434,8 @@ class RoomBattle extends RoomGames.RoomGame {
 		 * @type {number}
 		 */
 		this.rated = options.rated || 0;
+		// true when players have taken all open slots in an imported battle
+		this.missingPlayers = !!options.inputLog;
```

Comment:
Wait, why is this necessary?

Reply:
is this different from `!this.players.every(player => player.userid)`

Reply:
Yes, after players are restored in an input log, we need to display the battle start message that is called in `Rooms.global.onCreateBattleRoom`. This flag is so that we only display that message once when a player is added.

This could still be simplified if `/addplayer` isn't supposed to support replacing players.

Reply:
This should probably get a better name, then. `.missingBattleStartMessage`?